diagnostics/mem.go

```diff
@@ -0,0 +1,38 @@
+package diagnostics
+
+import (
+	"encoding/json"
+	"github.com/ledgerwatch/erigon-lib/common/dbg"
+	"net/http"
+	"time"
+)
+
+func SetupMemAccess(metricsMux *http.ServeMux) {
+	metricsMux.HandleFunc("/mem", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Content-Type", "application/json")
+		writeMem(w)
+	})
+
+	// update prometheus memory stats at least every 30 seconds
```

Comment:
this goroutine should maybe be started in the node init code instead of diagnostics - im not very familiar with diagnostics but I imagine that if we are running a node without diagnostics turned on then these metrics wont be updated?

Reply:
Moved it so the goroutine runs regardless of whether metrics is enabled.